shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/props/DataSourceProperties.java

```diff
@@ -119,4 +122,84 @@ public int hashCode() {
+        if (!parseJdbcUrl().contains("jdbc:mysql")) {
+            return;
+        }
+        for (Entry<String, Object> entry : getAllLocalProperties().entrySet()) {
+            if (!dataSourceProperties.getAllLocalProperties().containsKey(entry.getKey())) {
+                String msg = "target data source item:" + entry.getKey() + ", not in source data source:" + dataSourceProperties.getAllLocalProperties();
+                throw new InvalidDataSourcePropertiesException(dataSourceProperties.getDataSourceClassName(), msg);
+            }
+            if (entry.getKey().equals("jdbcUrl") && !isInSameDatabaseInstance(String.valueOf(entry.getValue()),
+                    String.valueOf(dataSourceProperties.getAllLocalProperties().get(entry.getKey())))) {
+                String msg = "target data source item:" + entry.getValue() + ", not equal source in data source:" + dataSourceProperties.getAllLocalProperties().get(entry.getKey());
+                throw new InvalidDataSourcePropertiesException(dataSourceProperties.getDataSourceClassName(), msg);
+            }
+            if (!entry.getKey().equals("jdbcUrl") && !String.valueOf(entry.getValue()).equals(String.valueOf(dataSourceProperties.getAllLocalProperties().get(entry.getKey())))) {
+                String msg = "target data source item: " + entry.getKey() + ", value:" + entry.getValue() + ", not equal source item: " + entry.getKey() + ",data source:"
+                        + dataSourceProperties.getAllLocalProperties().get(entry.getKey());
+                throw new InvalidDataSourcePropertiesException(dataSourceProperties.getDataSourceClassName(), msg);
+            }
+        }
+    }
+
+    /**
+     * Is data source in same instance.
+     *
+     * @param sourceUrl source url
+     * @param targetUrl target url
+     * @return boolean
+     */
+    public boolean isInSameDatabaseInstance(final String sourceUrl, final String targetUrl) {
+        if (sourceUrl.equals(targetUrl)) {
+            return true;
+        }
+        JdbcUrl sourceJdbcUrl = parseStandardJdbcUrlParserByJdbcUrl(sourceUrl);
+        JdbcUrl targetJdbcUrl = parseStandardJdbcUrlParserByJdbcUrl(targetUrl);
+        return sourceJdbcUrl.getHostname().equals(targetJdbcUrl.getHostname()) && sourceJdbcUrl.getPort() == targetJdbcUrl.getPort()
+                && sourceJdbcUrl.getQueryProperties().equals(targetJdbcUrl.getQueryProperties());
+    }
+
+    private JdbcUrl parseStandardJdbcUrlParserByJdbcUrl(final String jdbcUrl) {
+        StandardJdbcUrlParser jdbcUrlParser = new StandardJdbcUrlParser();
+        return jdbcUrlParser.parse(jdbcUrl);
+    }
+
+    private String parseJdbcUrl() {
+        for (Entry<String, Object> entry : getAllLocalProperties().entrySet()) {
+            if (entry.getKey().equals("jdbcUrl")) {
+                return String.valueOf(entry.getValue());
+            }
+        }
+        return "";
+    }
+
+    /**
+     * get instance from data source.
```

Comment:
First letter of java doc should be upper case

Reply:
Fix in https://github.com/apache/shardingsphere/pull/16632